### PR TITLE
feat: Support API token for scanning `hf://`

### PR DESCRIPTION
**Pull request by nameexhaustion**
Adds support for scanning from private repositories with an API token:

```python
pl.scan_csv(
    "hf://...",
    storage_options={"token": "..."},
)
```
or alternatively, using an environment variable
```shell
export HF_TOKEN=...
python main.py
```

ref https://github.com/pola-rs/polars/issues/17625

### CODE REVIEW

**Comment by albertvillanova** [comment]
Thanks for the fast implementation.

Just a nit comment: do you think the parameter could be named just "token"? It seems a bit redundant to call it "hf_token" when it is already passed to the "hf://" URL...

**Comment by ritchie46** [comment]
Can we add an entry in the docstring for discoverability? (Aside from the user guide)

File: py-polars/polars/io/parquet/functions.py [hunk 1]
```diff
@@ -59,12 +59,14 @@ def read_parquet(
     Parameters
     ----------
     source
-        Path to a file or a file-like object (by "file-like object" we refer to objects
+        Path(s) to a file or directory
+        When needing to authenticate for scanning cloud locations, see the `storage_options`
```
Comment (nameexhaustion) [inline py-polars/polars/io/parquet/functions.py:63]: update docs for `source` to point to cloud options

File: py-polars/src/lazyframe/mod.rs [hunk 2]
```diff
@@ -158,7 +158,7 @@ impl PyLazyFrame {
             let mut cloud_options = if let Some(opts) = cloud_options {
                 parse_cloud_options(&first_path_url, opts)?
             } else {
-                Default::default()
+                parse_cloud_options(&first_path_url, vec![])?
```
Comment (nameexhaustion) [inline py-polars/src/lazyframe/mod.rs:161]: the logic for loading HF_TOKEN from the environment requires this function call

File: py-polars/polars/io/parquet/functions.py [hunk 3]
```diff
@@ -365,6 +368,8 @@ def scan_parquet(
         * `aws <https://docs.rs/object_store/latest/object_store/aws/enum.AmazonS3ConfigKey.html>`_
         * `gcp <https://docs.rs/object_store/latest/object_store/gcp/enum.GoogleConfigKey.html>`_
         * `azure <https://docs.rs/object_store/latest/object_store/azure/enum.AzureConfigKey.html>`_
+        * Hugging Face (`hf://`): Accepts an API key under the `token` paramter: \
+          `{'token': '...'}`, or by setting the `HF_TOKEN` environment variable.
```
Comment (nameexhaustion) [inline py-polars/polars/io/parquet/functions.py:372]: added mention to HF token under `storage_options`

**Pull request merged**